lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -815,13 +824,24 @@ static void optimizeBatchNorm(Function *F) {
       // Q = W * A
       // C = b * A + B
 
-      auto filterH = cast<Variable>(CV->getFilter())->getHandle<>();
-      auto cbiasH = cast<Variable>(CV->getBias())->getHandle<>();
+      Variable *filterV = getModifiableVariable(*CV->getFilter());
+      Variable *cbiasV = getModifiableVariable(*CV->getBias());
+
+      Variable *scaleV = dyn_cast<Variable>(BN->getScale());
+      Variable *biasV = dyn_cast<Variable>(BN->getBias());
+      Variable *meanV = dyn_cast<Variable>(BN->getMean());
+      Variable *var = dyn_cast<Variable>(BN->getVar());
```

Comment:
based on the way we build batch normalization node we expect all those 4 be a variable, maybe place an assert on those instead of if checking.
This way if it changes we could adopt implementation of optimization otherwise this optimization will not trigger at all (would be easy to skip).

Reply:
That's the thing, this is going to change, because only load and save would be able to work directly with variables.

Reply:
I'll go back to the cast, because I'll have to change this code anyway when the change for loads land. It used to be consistent when I was using getModifiableVariable for all, but now that I rolled back to a straight dyn_cast, that doesn't make much sense.

Anyway, reverting to the straight cast here.